### PR TITLE
S3Client now use buffered stream

### PR DESCRIPTION
**Pull request by pzartem**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module github.com/FindHotel/analytics-go
+
+go 1.14
```
Comment (pavel-popov) [inline go.mod:3]: Maybe upgrade to 1.15?

File: s3client.go [hunk 2]
```diff
@@ -292,4 +306,172 @@ func (c *s3Client) upload(r io.Reader) error {
...
+	if q.messages >= q.maxBatchSize {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (q *bufferedEncoder) Reset() error {
+	err := q.buf.Reset()
+	if err != nil {
```
Comment (pavel-popov) [inline s3client.go:366]:
```suggestion
	if err := q.buf.Reset(); err != nil {
```

**Review by velppa — APPROVED** [review]
Great job! 🚀

File: analytics.go [hunk 3]
```diff
@@ -429,3 +435,17 @@ func (c *client) notifyFailure(msgs []message, err error) {
 		}
 	}
 }
+
+func (c *client) notifyFailureMsg(m Message, err error, count int64) {
```
Comment (pavel-popov) [inline analytics.go:439]: Are we sure we need those? Do we have a single client using it?

**Review by velppa — APPROVED** [review]
Great job! Thank you!

**Pull request merged**